### ISSUE TITLE
`dart fix` sometimes doesn't remove all redundant arguments for avoid_redundant_argument_values

### ISSUE THREAD

**Issue by Hixie** [dart-lang/sdk#47398]
Sometimes you have to run `dart fix` several times to get to to a steady state.

STEPS TO REPRODUCE
* `dart create dart_fix_test`
* Replace the `dart_fix_test/bin/dart_fix_test.dart` file as follows:

```dart
void main(List<String> arguments) {
  DateTime(1, 1, 1);
}
```

* Replace the analysis_options.yaml file as follows:

```yaml
linter:
  rules:
    - avoid_redundant_argument_values
```

* Run `dart fix --apply`

EXPECTED RESULTS

The file should look as follows:

```dart
void main(List<String> arguments) {
  DateTime(1);
}
```

...and running `dart fix --apply` again should have no effect.

ACTUAL RESULTS

```dart
void main(List<String> arguments) {
  DateTime(1, 1);
}
```

Running `dart fix --apply` a second time with no intermediate changes causes more fixes to be applied.

**Comment by Hixie** [comment]
Another case where it doesn't remove everything:

```dart
void test({ String? a, String? b, String? c });

void main(List<String> arguments) {
  test(a: null, b: null, c: null);
}
```

Running it once only removes `a` and `c`, but leaves `b`.

**Comment by scheglov** [comment]
@bwilkerson

**Comment by bwilkerson** [comment]
Yes, the `dart fix` command currently makes a single pass over the code fixing as many diagnostics as possible. Our longer term plans are to make it iterative. One of the first steps toward that goal is likely to be to add a final pass that will remove and reorder directives. (It's my belief that waiting to clean up directive until after other edits are computed will result in a better final result.) I'm hoping to be looking into that fairly soon.

The issue, however, isn't an example of that problem; it's an example of a failure to consider comments when computing edits.
